### PR TITLE
Signal-based, nice side menus

### PR DESCRIPTION
**Pull request by pferreir**

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Let's decide on whether to use lists or tuples for `__all__` - right now we do both..

File: indico/web/menu.py [hunk 1]
```diff
@@ -84,3 +86,94 @@ def __repr__(self):
...
+    Build a menu (list of entries) with sections/items.
+
+    Information is provided by specific signals and filtered
+    by menu id.
+    This can be used as a very thin framework for menu
+    handling across the app.
+
+    :param menu_id: menu_id used to filter out signal calls
+    :param active_item: ID of currently active menu item
+    :returns: properly sorted list (taking weights into account)
```
Comment (ThiefMaster) [inline indico/web/menu.py]:
```
$ ack -hc ':return:'
44
$ ack -hc ':returns:'
20
```

(both are correct)

File: indico/web/menu.py [hunk 2]
```diff
@@ -84,3 +86,94 @@ def __repr__(self):
...
+        self.active = active
+        self.disabled = disabled
+        self.visible = visible
+        self.event_feature = event_feature
+        self.section = section
+        self.weight = weight
+        self.icon = ('icon-' + icon) if icon else None
+
+
+def menu_from_signals(menu_id, active_item=None, **kwargs):
```
Comment (ThiefMaster) [inline indico/web/menu.py]: is the name still a good fit? (it builds it from signals, but that's more of an implementation detail)

File: indico/web/menu.py [hunk 3]
```diff
@@ -84,3 +86,94 @@ def __repr__(self):
...
+    """
+    Build a menu (list of entries) with sections/items.
+
+    Information is provided by specific signals and filtered
+    by menu id.
+    This can be used as a very thin framework for menu
+    handling across the app.
+
+    :param menu_id: menu_id used to filter out signal calls
+    :param active_item: ID of currently active menu item
```
Comment (ThiefMaster) [inline indico/web/menu.py]: `:param kwargs: extra arguments passed to the signals`

File: indico/web/menu.py [hunk 4]
```diff
@@ -84,3 +86,94 @@ def __repr__(self):
...
+        return self._active or any(item.active for item in self._items)
+
+
+class SideMenuItem(object):
+    """Defines a side menu item.
+
+    :param title: the title of the menu item (displayed)
+    :param url: the URL that the link will point to
+    :param weight: the "weight" (higher means it shows up first)
+    :param active: whether the item will be shown as active by default
```
Comment (ThiefMaster) [inline indico/web/menu.py]: `disabled` is missing here

File: indico/MaKaC/webinterface/pages/admins.py [hunk 5]
```diff
@@ -70,80 +72,25 @@ def _getHeader( self ):
...
-                                 key=lambda x: x[1]._title):
-            self.extra_menu_items[name] = item
-            self._sideMenu.addItem(item)
-
-    def _getBody( self, params ):
-        self._createSideMenu()
-        self._setActiveSideMenuItem()
+    def _getAdminSideMenu(self):
+        menu = menu_from_signals('admin-sidemenu',
+                                 active_item=self._get_active_menu_item())
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/admins.py]: fits in one line

File: indico/MaKaC/webinterface/pages/admins.py [hunk 6]
```diff
@@ -70,80 +72,25 @@ def _getHeader( self ):
...
-        self.extra_menu_items = {}
-        for name, item in sorted(values_from_signal(signals.admin_sidemenu.send()),
-                                 key=lambda x: x[1]._title):
-            self.extra_menu_items[name] = item
-            self._sideMenu.addItem(item)
-
-    def _getBody( self, params ):
-        self._createSideMenu()
-        self._setActiveSideMenuItem()
+    def _getAdminSideMenu(self):
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/admins.py]: what about adding a `render_menu(menu_id, active_item=None, old_style=False, **kwargs)` method that builds the menu and renders the template? then this method could go away and you could just call it directly in `_getBody`

File: indico/MaKaC/webinterface/pages/admins.py (hunk 5 above)
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/admins.py]: Why not use `self.sidemenu_option` directly? If it has to be dynamic you can always override the class-level attribute with a property (since you always access it via `self` - and a classproperty would probably work, too)

File: indico/core/signals/__init__.py [hunk 7]
```diff
@@ -14,5 +14,7 @@
 # You should have received a copy of the GNU General Public License
 # along with Indico; if not, see <http://www.gnu.org/licenses/>.
 
-from indico.core.signals import acl, agreements, attachments, category, event, event_management, plugin, users
+from indico.core.signals import (acl, agreements, attachments, category, event, event_management, menu, plugin, users)
```
Comment (ThiefMaster) [inline indico/core/signals/__init__.py]: why the `()` if it still fits in one line?

File: indico/core/signals/__init__.py [hunk 8]
```diff
@@ -14,5 +14,7 @@
...
 from indico.core.signals.core import *
+
+__all__ = ('acl', 'agreements', 'attachments', 'category', 'event', 'event_management', 'menu', 'plugin', 'users', 'rb')
```
Comment (ThiefMaster) [inline indico/core/signals/__init__.py]: at least here I think `__all__` is not a good idea - if you add a new signal e.g. to signals.core.py chances are good that you don't look at this file (and repeating all the core signals here feels ugly in any case)

actually all the core signals are missing in `__all__` here right now

File: indico/core/signals/menu.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+
+from blinker import Namespace
+
+_signals = Namespace()
+
+
+items = _signals.signal('menu-items', """
+Expected to return `(menu_item_name, SideMenuItem)` tuples to be added to
```
Comment (ThiefMaster) [inline indico/core/signals/menu.py]: If you used `named_object_from_signal` instead of `values_from_signal` and added a `name` attribute to the menu sections/items you could simply return/yield items instead of tuples. Also, you'd get a check to ensure unique names for free ;)

**Comment by pferreir** [comment]
I don't see any reason not to use tuples.

File: indico/web/menu.py (hunk 1 above)
Comment (pferreir) [inline indico/web/menu.py]: Maybe we can add a mass-commit that standardises everything to one of them?
But, OTOH, it's one thing that is easy to miss... if someone one of these days adds a `:returns:` then it will spread like wild fire again. I think this is really a case in which it won't hurt to be tolerant.

File: indico/htdocs/js/indico/modules/cephalopod.js [hunk 10]
```diff
@@ -71,7 +71,7 @@ function initCephalopdOnSettingsPage(cephalopodUrl) {
     if (!enabled) {
         defer.resolve({
             'title'   : $T('Disabled'),
-            'subtitle': $T('Your are not part of the community.'),
+            'subtitle': $T('You are not registered as part of the community.'),
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/cephalopod.js]: "Your instance is not as part of the community"?

File: indico/web/menu.py (hunk 2 above)
Comment (pferreir) [inline indico/web/menu.py]: What would you call it? I thought of `collect_menu_items`, but it doesn't seem that great. Something like `build_menu` would be too generic.

File: indico/web/menu.py (hunk 3 above)
Comment (pferreir) [inline indico/web/menu.py]: :+1:

File: indico/web/menu.py (hunk 2 above)
Comment (ThiefMaster) [inline indico/web/menu.py]: `build_menu_structure`? I guess it'd be used only in here anyway if there was a [`render_menu` function](https://github.com/indico/indico/pull/2064/files#r39760215)

File: indico/web/menu.py (hunk 4 above)
Comment (pferreir) [inline indico/web/menu.py]: :+1:

File: indico/modules/__init__.py [hunk 11]
```diff
@@ -15,4 +15,7 @@
 # along with Indico; if not, see <http://www.gnu.org/licenses/>.
 
 from indico.modules.base import ModuleHolder, Module
-import indico.modules.categories  # TODO: remove this import once it's imported from somewhere else
+
+# TODO: remove imports below once they're imported from somewhere else:
+import indico.modules.categories
```
Comment (ThiefMaster) [inline indico/modules/__init__.py]: c > a

File: indico/MaKaC/webinterface/pages/admins.py (hunk 5 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/admins.py]: Very good point.

File: indico/core/signals/__init__.py (hunk 7 above)
Comment (pferreir) [inline indico/core/signals/__init__.py]: It didn't fit in a previous lifetime, of course :wink:

File: indico/modules/categories/__init__.py [hunk 12]
```diff
@@ -64,3 +66,14 @@ def _redirect_legacy_id():
 
         request.view_args['categId'] = unicode(mapping.category_id)
         return redirect(url_for(request.endpoint, **dict(request.args.to_dict(), **request.view_args)), 301)
+
+
+@signals.menu.items.connect_via('category-management-sidemenu')
+def _sidemenu_items(sender, category, **kwargs):
+    yield 'view', SideMenuItem(_('View category'), url_for('category.categoryDisplay', category), 100, icon='eye')
```
Comment (ThiefMaster) [inline indico/modules/categories/__init__.py]: Always wrapping before the weight argument might be more readable

File: indico/modules/events/__init__.py [hunk 13]
```diff
@@ -59,6 +64,90 @@ def _event_deleted(event, **kwargs):
...
+    rb_active = Config.getInstance().getIsRoomBookingActive()
+    cfa_enabled = event.hasEnabledSection('cfa')
+    can_manage_registration = can_modify or event.canManageRegistration(session.user)
+    reg_form_enabled = event.hasEnabledSection('regForm')
+
+    event_type = event.getType()
+    is_lecture = event_type == 'simple_event'
+    is_conference = event_type == 'conference'
+    paper_review = event.getConfPaperReview()
+    is_review_staff = paper_review.isInReviewingTeam(session.user)
```
Comment (ThiefMaster) [inline indico/modules/events/__init__.py]: `session.avatar` or add `@unify_user_args(legacy=True)` on those two methods.

File: indico/core/signals/__init__.py (hunk 8 above)
Comment (pferreir) [inline indico/core/signals/__init__.py]: Yeah, I get it. But that's only for core signals. If you add a new signal file you'll have to come here.
Why can't we just add a `core` namespace, BTW? It seems weird to me that some signals are located one level above.

File: indico/modules/events/__init__.py [hunk 14]
```diff
@@ -59,6 +64,90 @@ def _event_deleted(event, **kwargs):
...
+        yield 'participants', SideMenuItem(_('Participants'),
+                                           url_for('event_mgmt.confModifParticipants', event),
+                                           section='organization',
+                                           visible=can_modify)
+
+    yield 'lists', SideMenuItem(_('Lists'),
+                                url_for('event_mgmt.confModifListings-allSpeakers', event),
+                                section='advanced',
+                                visible=(can_modify and not is_lecture))
+    yield 'utilities', SideMenuItem(_('Utilities'),
```
Comment (ThiefMaster) [inline indico/modules/events/__init__.py]: Separate section for the various utils?

File: indico/core/signals/menu.py (hunk 9 above)
Comment (pferreir) [inline indico/core/signals/menu.py]: Where is `named_object_from_signal ` even defined?

File: indico/core/signals/__init__.py (hunk 8 above)
Comment (ThiefMaster) [inline indico/core/signals/__init__.py]: Fine for me too. Before 2.0 we could move around some signals anyway (`signals.plugin.template_hook` is not only used for plugins for example)

File: indico/core/signals/menu.py (hunk 9 above)
Comment (ThiefMaster) [inline indico/core/signals/menu.py]: `named_objects_from_signal` (had a typo in the initial comment) is defined in `indico/util/signals.py`

File: indico/htdocs/js/indico/modules/cephalopod.js (hunk 10 above)
Comment (pferreir) [inline indico/htdocs/js/indico/modules/cephalopod.js]: I guess you mean "Your instance is not **registered** as part of the community"?
I'd rather say "This server is not registered as part of the community", as "instance" may seem a bit without a context.

**Comment by ThiefMaster** [comment]
Can we use the new menu system for the header menu, too? It's not a side menu but with a different template it might not be too hard..

File: indico/modules/__init__.py (hunk 11 above)
Comment (pferreir) [inline indico/modules/__init__.py]: Can't wait till we get rid of MaKaC, so that I start taking pre-commit PEP8 errors seriously...

File: indico/htdocs/js/indico/modules/cephalopod.js (hunk 10 above)
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/cephalopod.js]: yeah, that's what i meant and "server" sounds indeed better

File: indico/modules/categories/__init__.py (hunk 12 above)
Comment (pferreir) [inline indico/modules/categories/__init__.py]: Agreed!

File: indico/modules/events/__init__.py (hunk 14 above)
Comment (pferreir) [inline indico/modules/events/__init__.py]: Let's do that in a second commit, OK?

File: indico/web/menu.py (hunk 2 above)
Comment (pferreir) [inline indico/web/menu.py]: Seems fair enough, but doesn't really specify which kind of menu we're talking about. There are other kinds of menu that are at least for now not using this.

File: indico/modules/rb/__init__.py [hunk 15]
```diff
@@ -70,3 +72,42 @@ def _event_deleted(event, user, **kwargs):
...
+    yield 'book_room', SideMenuItem(_('Book a Room'), url_for('rooms.book'), 80, icon='checkmark')
+    yield 'map', SideMenuItem(_('Map of Rooms'), url_for('rooms.roomBooking-mapOfRooms'), 70, icon='location',
+                              visible=map_available)
+    yield 'calendar', SideMenuItem(_('Calendar'), url_for('rooms.calendar'), 60, icon='calendar')
+    yield 'my_bookings', SideMenuItem(_('My Bookings'), url_for('rooms.my_bookings'), 50, icon='time')
+    yield 'search_bookings', SideMenuItem(_('Search bookings'), url_for('rooms.roomBooking-search4Bookings'),
+                                          section='search')
+    yield 'search_rooms', SideMenuItem(_('Search rooms'), url_for('rooms.search_rooms'),
+                                       section='search')
+    yield 'book_in_my_rooms', SideMenuItem(_('Bookings in my rooms'), url_for('rooms.bookings_my_rooms'),
```
Comment (ThiefMaster) [inline indico/modules/rb/__init__.py]: `bookings_` would fit better - `book_` sounds like it's a page to book stuff

Also, those items used to be disabled when you don't own any rooms. Maybe we could hide the whole "my rooms" section if you don't manage any rooms?

File: indico/modules/rb/views/__init__.py [hunk 16]
```diff
@@ -51,128 +49,10 @@ def getCSSFiles(self):
...
-
-        self._leftMenu.addSection(self._roomsBookingOpt)
-        self._leftMenu.addSection(self._mapOpt)
-        self._leftMenu.addSection(self._calendarOpt)
-        self._leftMenu.addSection(self._bookingsOpt)
-        self._leftMenu.addSection(self._searchOpt)
-        self._leftMenu.addSection(self._roomsOpt)
-        self._leftMenu.addSection(self._blockingsOpt)
+        menu = menu_from_signals('rb-sidemenu',
+                                 active_item=self._get_active_menu_item())
```
Comment (ThiefMaster) [inline indico/modules/rb/views/__init__.py]: another one that fits in one line ;)

File: indico/modules/rb/views/user/reservations.py [hunk 17]
```diff
@@ -94,8 +89,10 @@ def __init__(self, rh, menu_item, **kwargs):
         self._menu_item = menu_item
```
Comment (ThiefMaster) [inline indico/modules/rb/views/user/reservations.py]: why not set `self.sidemenu_option`?

File: indico/modules/vc/views.py [hunk 18]
```diff
@@ -26,6 +26,7 @@ class WPVCJinjaMixin(WPJinjaMixin):
 
 
 class WPVCManageEvent(WPVCJinjaMixin, WPConferenceModifBase):
+    sidemenu_option = 'vc'
```
Comment (ThiefMaster) [inline indico/modules/vc/views.py]: `videoconference`? I kind of like `vc` but at least the conference menu entry already uses `videoconference` in its name

File: indico/MaKaC/webinterface/pages/admins.py (hunk 6 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/admins.py]: :+1:

File: indico/modules/events/__init__.py (hunk 13 above)
Comment (pferreir) [inline indico/modules/events/__init__.py]: :+1:

File: indico/modules/rb/views/user/reservations.py (hunk 17 above)
Comment (pferreir) [inline indico/modules/rb/views/user/reservations.py]: Overriding it directly seemed a bit confusing to me, as the value is coming from the RH. I though overriding it with a method/property and then returning the stored value would be more explicit and easier to understand. Not that sure now, though...

File: indico/modules/vc/views.py (hunk 18 above)
Comment (pferreir) [inline indico/modules/vc/views.py]: I don't want anyone to think we're getting venture capital.

File: indico/modules/rb/views/__init__.py (hunk 16 above)
Comment (pferreir) [inline indico/modules/rb/views/__init__.py]: Already fixed due to other changes.

File: indico/modules/rb/__init__.py (hunk 15 above)
Comment (pferreir) [inline indico/modules/rb/__init__.py]: I thought of using `bookings_`, but then `prebookings_xxxx_yyyy` would seem too long. Anyway, you're right, better spend some characters.

> Also, those items used to be disabled when you don't own any rooms. Maybe we could hide the whole "my rooms" section if you don't manage any rooms?

Had already fixed that, meanwhile :wink:

File: indico/core/signals/menu.py (hunk 9 above)
Comment (pferreir) [inline indico/core/signals/menu.py]: Erm... I guess I'll leave this for a future commit. We're just using the tuple pattern everywhere, not only in side menus...

Comment (ThiefMaster) [inline indico/core/signals/menu.py]: Mostly in places we added before I added `named_object_from_signal` - but yeah, makes sense to do it in a separate commit

**Comment by pferreir** [comment]
Summarizing what's left to do:
 * Moving things under "utilities" to separate options - should be discussed, can be done in separate commit
 * Moving head menu to this new architecture (and possible the conference display menu as well?)
 * Using `named_object_from_signal ` - can be done in a separate commit>

I'd go on with the first one, and create tickets for the other two. I know it's tempting to do them now, but I've already spent quite some time with this.

**Comment by ThiefMaster** [comment]
Do we actually need the `visible` attribute of menu items? Since it's a static value we could simply not return/yield the item at all from the signal if we don't want to show it.

File: indico/MaKaC/webinterface/pages/conferences.py [hunk 19]
```diff
@@ -1010,175 +1012,23 @@ def _getNavigationDrawer(self):
...
-    def _applyFrame( self, body ):
-        frame = wcomponents.WConferenceModifFrame( self._conf, self._getAW())
-
-        sideMenu = self._sideMenu.getHTML()
+        params = {
+            "categDisplayURLGen": urlHandlers.UHCategoryDisplay.getURL,
+            "confDisplayURLGen": urlHandlers.UHConferenceDisplay.getURL,
+            "event": "Conference",
+            "sideMenu": render_sidemenu('event-management-sidemenu', active_item=self.sidemenu_option, old_style=True,
+                                        event=self._conf)
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/conferences.py]: We could already use `self._conf.as_event` here and adapt the signals accordingly (`event.can_manage(session.user, allow_key=True)` for items that require full management access)

File: indico/MaKaC/webinterface/pages/conferences.py [hunk 20]
```diff
@@ -2697,8 +2480,7 @@ def getVars( self ):
 
 class WPConfModifProgram( WPConferenceModifBase ):
 
-    def _setActiveSideMenuItem( self ):
-        self._programMenuItem.setActive()
+    sidemenu_option = 'programme'
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/conferences.py]: program? (it's using the american spelling in both the WP/RH names and also in the conference sidemenu item name)

File: indico/MaKaC/webinterface/pages/contributions.py [hunk 21]
```diff
@@ -278,11 +278,12 @@ def _setActiveTab(self):
     def _setupTabCtrl(self):
         pass
 
-    def _setActiveSideMenuItem(self):
+    @property
+    def sidemenu_option(self):
         if self._target.isScheduled():
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/contributions.py]: ternary if..else?

File: indico/MaKaC/webinterface/pages/main.py [hunk 22]
```diff
@@ -25,22 +25,17 @@
...
+        target = self._rh.getTarget()
 
         # Check if user is administrator
         self._isAdmin = session.user and session.user.is_admin
-        self._isCategoryManager = self._isAdmin or \
-                                  isinstance(self._rh.getTarget(), Category) and self._rh.getTarget().canModify(self._getAW()) or \
-                                  isinstance(self._rh.getTarget(), Conference) and self._rh.getTarget().getOwnerList()!=[] and self._rh.getTarget().getOwner().canModify(self._getAW())
+        self._isCategoryManager = (
+            self._isAdmin or
+            isinstance(target, Category) and target.canModify(self._getAW()) or
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/main.py]: I know you just reformatted it, but I'd add parentheses around the OR'd items which contain ANDs

File: indico/MaKaC/webinterface/pages/main.py [hunk 23]
```diff
@@ -48,12 +43,15 @@ def _display( self, params ):
...
                          isFrontPage=self._isFrontPage(),
                          isRoomBooking=self._isRoomBooking(),
-                         sideMenu=sideMenu).getHTML({"subArea": self._getSiteArea()})
+                         sideMenu=self._getSideMenu()).getHTML({"subArea": self._getSiteArea()})
 
         return self._applyDecoration(body)
 
     def _getBody(self, params):
-        return _("nothing yet")
+        return ''
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/main.py]: Should we raise an exception instead? It's pretty much an error case if a page doesn't have any content

File: indico/MaKaC/webinterface/pages/subContributions.py [hunk 24]
```diff
@@ -145,11 +145,12 @@ def _getPageContent( self, params ):
 
         return banner + body
 
-    def _setActiveSideMenuItem(self):
+    @property
+    def sidemenu_option(self):
         if self._contrib.isScheduled():
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/subContributions.py]: ternary?

File: indico/MaKaC/webinterface/tpls/CategModifMain.tpl [hunk 25]
```diff
@@ -1,3 +1,5 @@
+${ render_template('flashed_messages.html') }
```
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/tpls/CategModifMain.tpl]: Do we have ugly full-page-width flashes on that page? If yes maybe we could use `render_template('flashed_messages.html', fixed_width=true)` and add change flashed_messages.html like this: ` {{ message_box(category, message, fixed_width=fixed_width|default(false)) }}` (the default is not really needed, but I think passing false instead of Undefined is cleaner)

File: indico/htdocs/sass/modules/_event_management.scss [hunk 26]
```diff
@@ -391,3 +391,21 @@ form#upload-images.full-width .dropzone {
...
+
+    input[type="checkbox"] {
+        font-size: 2em;
+    }
+
+    label {
+        font-weight: normal;
+        margin-left: 0.5em;
+    }
+}
```
Comment (ThiefMaster) [inline indico/htdocs/sass/modules/_event_management.scss]: missing linebreak at EOF

File: indico/modules/attachments/__init__.py [hunk 27]
```diff
@@ -45,11 +45,10 @@ def _merge_users(target, source, **kwargs):
     AttachmentFolderPrincipal.merge_users(target, source, 'folder')
 
 
-@signals.event_management.sidemenu.connect
-def _extend_event_management_menu(event, **kwargs):
-    from MaKaC.webinterface.wcomponents import SideMenuItem
+@signals.menu.items.connect_via('event-management-sidemenu')
+def _extend_event_management_menu(sender, event, **kwargs):
     return 'attachments', SideMenuItem(_('Materials'), url_for('attachments.management', event),
```
Comment (ThiefMaster) [inline indico/modules/attachments/__init__.py]:
```
if not can_manage_attachments(event, session.user):
    return
```

File: indico/modules/events/__init__.py [hunk 28]
```diff
@@ -27,10 +27,12 @@
 from indico.modules.events.models.legacy_mapping import LegacyEventMapping
 from indico.modules.events.models.settings import EventSetting, EventSettingPrincipal
 from indico.modules.events.util import notify_pending
-from indico.web.flask.util import url_for
 from indico.util.i18n import _, ngettext, orig_string
 from indico.util.string import is_legacy_id
+from indico.web.flask.util import url_for
+
 
+__all__ = ('Event',)
```
Comment (ThiefMaster) [inline indico/modules/events/__init__.py]: 'event_management_object_url_prefixes', 'event_object_url_prefixes'

File: indico/modules/events/logs/__init__.py [hunk 29]
```diff
@@ -23,16 +23,16 @@
 from indico.modules.events.logs.renderers import SimpleRenderer, EmailRenderer
 from indico.modules.events.logs.util import get_log_renderers
 from indico.web.flask.util import url_for
+from indico.web.menu import SideMenuItem
 
-__all__ = ['EventLogEntry', 'EventLogKind']
+__all__ = ['EventLogEntry', 'EventLogKind', 'EventLogRealm']
```
Comment (ThiefMaster) [inline indico/modules/events/logs/__init__.py]: tuple

File: indico/modules/events/management/__init__.py [hunk 30]
```diff
@@ -0,0 +1,111 @@
...
+
+
+@signals.menu.items.connect_via('event-management-sidemenu')
+def _sidemenu_items(sender, event, **kwargs):
+
+    # Some legacy handling of menu items
+    # Once those parts are modularized, we will be able to include them
+    # conditionally from their respective modules.
+
+    can_modify = event.canModify(session.user)
```
Comment (ThiefMaster) [inline indico/modules/events/management/__init__.py]: I'd pass the new Event object to that signal and use `can_manage`

File: indico/modules/events/management/__init__.py [hunk 31]
```diff
@@ -0,0 +1,111 @@
...
+    rb_active = Config.getInstance().getIsRoomBookingActive()
+    cfa_enabled = event.hasEnabledSection('cfa')
+    can_manage_registration = can_modify or event.canManageRegistration(session.user)
```
Comment (ThiefMaster) [inline indico/modules/events/management/__init__.py]: `can_manage(session.user, 'registration')`

**Comment by ThiefMaster** [comment]
We have pretty ugly code in `RHConferenceModifManagementAccess` atm to determine the management page the user can access. Maybe we could change this to simply go to the the first menu item?

File: indico/modules/events/management/blueprint.py [hunk 32]
```diff
@@ -0,0 +1,32 @@
...
+from indico.modules.events.management.controllers import (RHDeleteEventAction, RHDeleteEventForm, RHLockEventAction,
+                                                          RHLockEventForm)
+from indico.web.flask.wrappers import IndicoBlueprint
+
+
+_bp = IndicoBlueprint('event_management', __name__, template_folder='templates',
+                      virtual_template_folder='events/management',
+                      url_prefix='/event/<confId>/manage')
+
+_bp.add_url_rule('/delete', 'delete', RHDeleteEventForm, methods=('GET',))
```
Comment (ThiefMaster) [inline indico/modules/events/management/blueprint.py]: GET is the default

File: indico/modules/events/management/controllers.py [hunk 33]
```diff
@@ -0,0 +1,70 @@
...
+    CSRF_ENABLED = True
+
+    def _process(self):
+        category = self._conf.getOwner()
+
+        self._conf.delete(session.user)
+        flash(_('Event "{}" successfully deleted.').format(self._conf.title), 'success')
+
+        if category:
+            redirect_url = url_for('category_mgmt.categoryModification', category)
```
Comment (ThiefMaster) [inline indico/modules/events/management/controllers.py]: hm, what about redirecting back to the normal category page? especially for big categories it's a bit ugly to end up on category management after deleting an event

File: indico/modules/events/management/controllers.py [hunk 34]
```diff
@@ -0,0 +1,70 @@
...
+from flask import flash, session
+
+from indico.util.i18n import _
+from indico.web.flask.util import url_for, jsonify_data
+from indico.web.util import jsonify_template
+
+from MaKaC.webinterface.rh.conferenceModif import RHConferenceModifBase
+
+
+class RHDeleteEventForm(RHConferenceModifBase):
```
Comment (ThiefMaster) [inline indico/modules/events/management/controllers.py]: I think form and execution could be in one RH, using `_process_GET` and `_process_POST`

File: indico/modules/events/management/controllers.py [hunk 35]
```diff
@@ -0,0 +1,70 @@
...
+
+
+class RHLockEventAction(RHConferenceModifBase):
+    """Lock an event."""
+
+    CSRF_ENABLED = True
+
+    def _process(self):
+        self._conf.setClosed(True)
+        flash(_('The event is now locked.').format(self._conf.title), 'success')
```
Comment (ThiefMaster) [inline indico/modules/events/management/controllers.py]: missing placeholder for the title

File: indico/modules/events/management/templates/action_menu.html [hunk 36]
```diff
@@ -0,0 +1,37 @@
...
+           title="{% trans %}See the display page of the event{% endtrans %}"
+           href="{{ url_for('event.conferenceDisplay', event) }}">{% trans %}Switch to display view{% endtrans %}</a>
+    </div>
+</div>
+
+<script>
+    $('.js-ajax-dialog-delete').ajaxDialog({
+        onClose: function(data) {
+            if (data && data.url) {
+                location = data.url;
```
Comment (ThiefMaster) [inline indico/modules/events/management/templates/action_menu.html]: assigning it to `location.href` feels cleaner

File: indico/modules/events/management/templates/action_menu.html [hunk 37]
```diff
@@ -0,0 +1,37 @@
...
+            }
+            return true;
+        }
+    });
+
+    $('.js-ajax-dialog-lock').ajaxDialog({
```
Comment (ThiefMaster) [inline indico/modules/events/management/templates/action_menu.html]: same code as above -> could `js-ajax-dialog-delete, js-ajax-dialog-lock` and call ajaxDialog() only once

File: indico/modules/events/management/templates/delete_event.html [hunk 38]
```diff
@@ -0,0 +1,42 @@
...
+                <button id="js-button-close" class="i-button big right" data-button-back>
+                    {% trans %}I've changed my mind!{% endtrans %}
+                </button>
+            </div>
+        </div>
+    </form>
+
+    <script>
+        $('#js-confirm-delete').on('change', function() {
+            $('#js-button-delete').prop('disabled', !$(this).is(':checked'));
```
Comment (ThiefMaster) [inline indico/modules/events/management/templates/delete_event.html]: `!this.checked`

File: indico/modules/events/management/templates/delete_event.html [hunk 39]
```diff
@@ -0,0 +1,42 @@
...
+
+{% block content %}
+    <form method="POST" class="i-form vertical js-form-confirm-delete">
+        <input type="hidden" name="csrf_token" value="{{ session.csrf_token }}">
+        <h2>{{ event.title }}</h2>
+        <div class="form-group danger-message-box large-icon">
+            <div class="message-text">
+                {% trans %}
+                    Please note that if you delete the event you will lose all the information contained
+                    in it. <br/><strong>This operation is irreversible!</strong>
```
Comment (ThiefMaster) [inline indico/modules/events/management/templates/delete_event.html]: `<br>` and I'd add a linebreak after it

File: indico/MaKaC/webinterface/pages/main.py (hunk 22 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/main.py]: Really? Isn't it common knowledge that `and` has precedence?

File: indico/MaKaC/webinterface/pages/main.py (hunk 23 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/main.py]: Hm.... I'm gonna raise `NotImplemented`, I think it make sense.

File: indico/MaKaC/webinterface/pages/main.py (hunk 22 above)
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/main.py]: > Explicit is better than implicit.

;)

File: indico/MaKaC/webinterface/pages/main.py (hunk 23 above)
Comment (ThiefMaster) [inline indico/MaKaC/webinterface/pages/main.py]: `NotImplementedError` - `NotImplemented` is meant for places where you return it (e.g. comparison functions)

File: indico/MaKaC/webinterface/tpls/CategModifMain.tpl (hunk 25 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/tpls/CategModifMain.tpl]: It doesn't actually look **that** ugly. And `fixed_width=True` doesn't make it look that great either. But yeah, I'll add it for the sake of future usage.

File: indico/MaKaC/webinterface/pages/conferences.py (hunk 20 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/conferences.py]: Change done.

File: indico/MaKaC/webinterface/pages/contributions.py (hunk 21 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/contributions.py]: Yeah, that was already like that. But I'm changing it nonetheless.

File: indico/MaKaC/webinterface/pages/main.py (hunk 22 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/main.py]: Well, then you would never use list comprehension :wink: 
But whatever, I'll add parenthesis for the less experienced.

File: indico/MaKaC/webinterface/pages/main.py (hunk 23 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/main.py]: :+1:

File: indico/MaKaC/webinterface/pages/subContributions.py (hunk 24 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/subContributions.py]: Same.

File: indico/modules/attachments/__init__.py (hunk 27 above)
Comment (pferreir) [inline indico/modules/attachments/__init__.py]: You had missed that one :wink:

File: indico/MaKaC/webinterface/pages/conferences.py (hunk 19 above)
Comment (pferreir) [inline indico/MaKaC/webinterface/pages/conferences.py]: Well, hoping you won't change any other APIs in the meantime.

File: indico/modules/events/management/controllers.py (hunk 34 above)
Comment (pferreir) [inline indico/modules/events/management/controllers.py]: That's not that simple. Don't forget it's inheriting from `RHConferenceModifBase`.

File: indico/modules/events/management/controllers.py (hunk 35 above)
Comment (pferreir) [inline indico/modules/events/management/controllers.py]: Rather "copy-paste" fail... we don't need a title there.

File: indico/modules/events/management/controllers.py (hunk 34 above)
Comment (ThiefMaster) [inline indico/modules/events/management/controllers.py]: Ah... the rather stupid default `_process` in one of the base conference RHs which I believe is only used for one page? :/

**Comment by ThiefMaster** [comment]
Merged a6dd62a
